middlewares/correlation_id_header.go

```diff
@@ -0,0 +1,47 @@
+package middlewares
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gofrs/uuid"
+)
+
+const CorrelationIDKey = "correlation-id"
+
+var correlationIDHeaders = []string{"X-Correlation-ID", "X-CorrelationID", "X-ForRequest-ID", "X-Request-ID", "X-Vcap-Request-Id"}
```

Comment:
Is it necessary to verify all of these headers? If so, could you add tests for the missing ones (a [table test](https://onsi.github.io/ginkgo/#table-driven-tests) or a for loop would do it)?

 Do we care if more than one is passed?

Reply:
Yes, you are absolutely right, there should be tests for each header. About the other question I don't think there should ever be more than one header, that carries such meaning for a request. But if somehow there is more than one header this indicates an integration problem between system components. We check more than one header because different systems can set the correlation id in different headers and these are the ones that are commonly used. The first one we find will be read, the rest will be ignored.